lib/filter/enemy.js

```diff
@@ -83,12 +83,49 @@ exports.update = function(json) {
 
         var buffs = '';
         if (config.get('enemy.defAttributes.enabled')) {
-          currentEnemy.params[0].def_attributes = config.get('enemy.defAttributes.attributes');
-
+          var NewDefAttrib = [];
+          var DisplayAttrib = [];
+          for (var att in json[round].enemy[enemy].children[part].params[0].def_attributes) {
+            var attributes = json[round].enemy[enemy].children[part].params[0].def_attributes[att]
+            if (attributes.attribute_id < 200 && attributes.factor == 1) {
+              NewDefAttrib.push(attributes);
+              switch (attributes.attribute_id) {
+              	case "100":
+                  DisplayAttrib.push("Fire");
+                  break;
+                case "101":
+                  DisplayAttrib.push("Ice");
+                  break;
+                case "102":
+                  DisplayAttrib.push("Thunder");
+                  break;
+                case "103":
+                  DisplayAttrib.push("Earth");
+                  break;
+                case "104":
+                  DisplayAttrib.push("Wind");
+                  break;
+                case "105":
+                  DisplayAttrib.push("Water");
+                  break;
+                case "106":
+                  DisplayAttrib.push("Holy");
+                  break;
+                case "107":
+                  DisplayAttrib.push("Dark");
+                  break;
+                default:
+                  DisplayAttrib.push("Poison");
```

Comment:
No need to declare a case and then have no default. Since in order to trigger the switch we have to actually have one of the 8 attributes, it's either one of the 7 before it and if not - it's poison.